client/src/js/controller.js

```diff
@@ -94,10 +72,68 @@ module.exports = function (app) {
         // setup scope and handlers
         controllerImplementation();
 
-        return;
+        $scope.$root.$on('onModelChanged', function (e, data) {
+          currHash = data.hash;
 
-        // business logic
+          tempHash = currHash;
+
+          window.location.hash = currHash.replace(/\//g, "_slash_").replace(/%2F/g, "_slash_");
```

Comment:
repetition of functionality `currHash.replace(/\//g, "_slash_").replace(/%2F/g, "_slash_")`